source/features/quick-reactions.tsx

```diff
@@ -0,0 +1,98 @@
+import delegate, {DelegateEvent} from 'delegate-it';
+import React from 'dom-chef';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+
+const emojiSortOrder = ['THUMBS_UP', 'THUMBS_DOWN', 'LAUGH', 'HOORAY', 'CONFUSED', 'HEART', 'ROCKET', 'EYES'];
+
+function optimisticReactionFromPost(event: DelegateEvent<MouseEvent>): void {
+	const reaction = event.delegateTarget as HTMLButtonElement;
+	updateReaction(reaction);
+}
+
+function optimisticReactionFromMenu(event: DelegateEvent<MouseEvent>): void {
+	const reactionButton = event.delegateTarget as HTMLButtonElement;
+	const reactionsMenu = reactionButton.closest('reactions-menu')!;
+	const details = reactionsMenu.querySelector('details');
+	details!.open = false; // Close reactions menu immediately
+
+	const reactionsContainer = reactionsMenu.nextElementSibling!.querySelector('.js-comment-reactions-options')!;
+	const reactionButtonValue = reactionButton.getAttribute('value')!;
+	const existingReaction = reactionsContainer.querySelector<HTMLButtonElement>(`[value="${reactionButtonValue}"]`);
+
+	if (existingReaction) {
+		// The user is updating an existing reaction via the menu, just update the reaction
+		updateReaction(existingReaction);
+	} else {
+		// Reactions have a specific order, so we need to insert the new reaction in the correct position
```

Comment:
If:

- GitHub still updates the UI later
- it simplifies this piece of code

Please just **append** the reaction instead, it will update within a second anyway. I'd rather have a feature that lasts longer than one that looks _absolutely perfect_ for only one month.